internal/scm/github/github.go

```diff
@@ -213,22 +218,34 @@ func (g *Github) getRepositories(ctx context.Context) ([]*github.Repository, err
 		allRepos = append(allRepos, repo)
 	}
 
-	// Remove duplicate repos
-	repoMap := map[string]*github.Repository{}
-	for _, repo := range allRepos {
-		repoMap[repo.GetFullName()] = repo
+	if len(g.RepositorySearch) > 0 {
+		repos, err := g.getSearchRepositories(ctx, g.RepositorySearch)
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not get repository search results for '%s'", g.RepositorySearch)
+		}
+		allRepos = append(allRepos, repos...)
 	}
-	allRepos = make([]*github.Repository, 0, len(repoMap))
-	for _, repo := range repoMap {
-		if repo.GetArchived() || repo.GetDisabled() {
-			continue
+	// Remove duplicate repos
+	var state string // Will be used to detect duplicates
```

Comment:
I think you should use a `map` here to keep track of duplicates. The double-sorting is kinda ugly. And using `map` for duplicate detection is totally acceptable in go.